hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactOperator.java

```diff
@@ -95,11 +101,32 @@ public void open() throws Exception {
       this.executor = NonThrownExecutor.builder(LOG).build();
     }
     this.collector = new StreamRecordCollector<>(output);
+    this.mutex = new Object();
+  }
+
+  @Override
+  public void processWatermark(Watermark mark) throws Exception {
+    // thread safe to propagate the watermark when asyncCompaction is enabled
+    if (asyncCompaction) {
+      synchronized (mutex) {
+        super.processWatermark(mark);
+      }
+      return;
+    }
+    super.processWatermark(mark);
   }
 
   @Override
-  public void processWatermark(Watermark mark) {
-    // no need to propagate the watermark
+  public void processLatencyMarker(LatencyMarker latencyMarker)
+      throws Exception {
+    // thread safe to propagate the latencyMarker when asyncCompaction is enabled
+    if (asyncCompaction) {
+      synchronized (mutex) {
+        super.processLatencyMarker(latencyMarker);
+      }
```

Comment:
Can we just ignore the watermark and latency marker totally ? The compaction operators belongs to the sink ppeline, there is no need to propagate them to downstream.

Reply:
In my opinion, it will be more user-friendly to display through flink ui after delivery, and some specific functions may need to deliver this information. So i do not  ignore the watermark and latency marker totally.

Reply:
now, i have  ignored the watermark and latency marker totally. @danny0405